crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -879,37 +879,43 @@ impl IndexerStore for PgIndexerStore {
         }).context(&format!("Failed reading transaction digests with checkpoint_sequence_number {checkpoint_sequence_number:?} and start_sequence {start_sequence:?} and limit {limit}"))
     }
 
-    async fn get_transaction_page_by_transaction_kind(
+    async fn get_transaction_page_by_transaction_kinds(
         &self,
-        kind: String,
+        kind_names: Vec<String>,
         start_sequence: Option<i64>,
         limit: usize,
         is_descending: bool,
     ) -> Result<Vec<Transaction>, IndexerError> {
-        read_only_blocking!(&self.blocking_cp, |conn| {
-            let mut boxed_query = transactions::dsl::transactions
-                .filter(transactions::dsl::transaction_kind.eq(kind.clone()))
-                .into_boxed();
+        if kind_names.is_empty() {
+            return Err(IndexerError::InvalidArgumentError(
+                "kind_names cannot be empty".to_string(),
+            ));
+        }
+
+        let kind_sub_query = kind_names
+            .iter()
+            .map(|kind_name| format!("transaction_kind = '{}'", kind_name))
```

Comment:
This seems vulnerable to a SQL injection attack.  Why the move away from the DSL (which I'm presuming/hoping protects us from SQL injection sttacks)?

Reply:
Yeah i think we could use [`eq_any`](https://docs.rs/diesel/latest/diesel/expression_methods/trait.ExpressionMethods.html#method.eq_any) instead of `eq` to handle the list of kinds